lib/HLSL/DxilValidation.cpp

```diff
@@ -3100,6 +3100,15 @@ static void ValidateFunctionBody(Function *F, ValidationContext &ValCtx) {
             }
             dispatchMesh = CI;
           }
+
+          for (Value *op : CI->operands()) {
```

Comment:
I don't think this is the right place for this code.

This should be implemented in `ValidateDxilOperationCallInProfile`, in the switch statement under the appropriate OpCodes.  See for instance how resource ops will use `ValidateResourceDxilOp(...)`.